File: lab6/Java/src/warehouse/resources/Ammunition.java
package warehouse.resources;

import enums.Caliber;
import enums.MeasureUnit;
import enums.SupplyType;
import java.util.Random;
import request.SupplyRequestDetail;

public class Ammunition extends Resource implements Cloneable{
    Caliber caliber;
    public Ammunition(int id, String name, Caliber caliber, int count)
    {
        super(id, name, MeasureUnit.ITEM, count);
        this.caliber = caliber;
    }

    @Override
    public Boolean isMatches(SupplyRequestDetail detail) {
        return detail.getSupplyType() == SupplyType.AMMUNITION &&
               detail.getCaliber() == caliber;
    }

    public Caliber getCaliber() {
        return caliber;
    }

    @Override
    public Ammunition clone() throws CloneNotSupportedException{
        var clone = (Ammunition)super.clone();
        clone.id = new Random().nextInt(0,9999);
        return clone;
    }
}

File: CourseProject/ILS_WPF/Models/Core/Resources/Ammunition.cs
using ILS_WPF.Models.Core.Enums;
using ILS_WPF.Models.Core.Requests;

namespace ILS_WPF.Models.Core.Resources
{
    public class Ammunition : Resource
    {
        public Caliber Caliber { get; set; }

        protected Ammunition() { }

        public Ammunition(string name, Caliber caliber, int quantity)
        {
            Name = name;
            MaterialType = MaterialType.Ammunition;
            Caliber = caliber;
            Quantity = quantity;
        }

        public override bool IsMatches(SupplyRequestDetail detail)
            => detail.MaterialType == MaterialType.Ammunition &&
                   Caliber == detail.Caliber;
    }
}
